src/commands/deploy.ts

```diff
@@ -64,43 +65,95 @@ export async function deployCommand(
   }
 
   const db = database.getConnection()
-  let deployCommands = fs
-    .readdirSync(outputFolderPath)
-    // Only extract .plv8.sql files, this will need to change if we ever make the extension configurable
-    .filter((file) => file.endsWith('.plv8.sql'))
-    .map((file) => {
-      const filePath = path.join(outputFolderPath, file)
-      return {
-        filePath,
-        sqlQueryPromise: db.file(filePath),
-      }
-    })
 
-  await task(
-    `Deploying files from ${outputFolderPath} to the provided PostgreSQL database 🚧`.trim(),
-    async ({ setWarning }) => {
-      const taskGroup = await task.group((task) =>
-        deployCommands.map((deployCommand) => {
-          const name = getFunctionNameFromFilePath(deployCommand.filePath)
-          return task(
-            `Deploying ${name}`,
-            async ({ setTitle: _setTitle, setError: _setError }) => {
-              try {
-                await deployCommand.sqlQueryPromise
-                _setTitle(`Deployed ${name}`)
-              } catch (e) {
-                _setError(`Failed to deploy ${name} (because of ${e.message})`)
-                setWarning(`Failed to some functions (see below))`)
-              }
-            }
+  await match(CLI.config.deployMode)
+    .with('functions', async () => {
+      const deployCommands = fs
+        .readdirSync(outputFolderPath)
+        // Only extract .plv8.sql files, this will need to change if we ever make the extension configurable
+        .filter((file) => file.endsWith('.plv8.sql'))
+        .map((file) => {
+          const filePath = path.join(outputFolderPath, file)
+          return {
+            filePath,
+            sqlQueryPromise: db.file(filePath),
+          }
+        })
+
+      await task(
+        `Deploying files from ${outputFolderPath} to the provided PostgreSQL database 🚧`.trim(),
+        async ({ setWarning }) => {
+          const taskGroup = await task.group((task) =>
+            deployCommands.map((deployCommand) => {
+              const name = getFunctionNameFromFilePath(deployCommand.filePath)
+              return task(
+                `Deploying ${name}`,
+                async ({ setTitle: _setTitle, setError: _setError }) => {
+                  try {
+                    await deployCommand.sqlQueryPromise
+                    _setTitle(`Deployed ${name}`)
+                  } catch (e) {
+                    _setError(
+                      `Failed to deploy ${name} (because of ${e.message})`
+                    )
+                    setWarning(`Failed to some functions (see below))`)
+                  }
+                }
+              )
+            })
           )
+
+          // TODO: add some batching here
+          await Promise.allSettled(taskGroup)
+        }
+      )
+    })
+    .with('pg_tle', async () => {
```

Comment:
Added two modes, 'pg_tle' mode gets the new code that instead of executing each individual SQL function file, collects them as a single string and executes `SELECT pgtle.install_extension` with appropriate parameters on it.